pkg/arvo/app/chat.hoon

```diff
@@ -340,201 +216,195 @@
+  =/  nes=(unit (list envelope:hall))  (~(get by messages) circle)
+  ?~  nes
+    [~ this]
+  =.  messages.str  (~(put by messages) circle (snoc u.nes nev.sto))
+  :_  this(str str)
+  (send-chat-update [[%message circle nev.sto] str])
+::
+++  handle-rumor-circle-status
+  |=  sto=rumor-story:hall
+  ^-  (quip move _this)
+  ?>  ?=(%status -.sto)
+  =/  upeers=(unit (set @p))  (~(get by peers.str) cir.sto)
+  ?~  upeers
+    [~ this]
+  =/  peers=(set @p)
+    ?:  =(%remove -.dif.sto)
+      (~(del in u.upeers) who.sto)
+    (~(put in u.upeers) who.sto)
+  =.  peers.str  (~(put by peers.str) cir.sto peers)
+  :_  this(str str)
+  (send-chat-update [[%peers cir.sto peers] str])
+::
+++  handle-rumor-circle-config-read
+  |=  sto=rumor-story:hall
+  ^-  (quip move _this)
+  ?>  ?=(%config -.sto)
+  ?>  ?=(%read -.dif.sto)
+  ?:  =(cir.sto [our.bol %inbox])
+    ::  ignore when cir.sto is inbox
+    [~ this]
+  =/  conf=(unit config:hall)  (~(got by configs.str) cir.sto)
+  ?~  conf
+    [~ this]
+  =.  red.u.conf  red.dif.sto
+  =.  configs.str  (~(put by configs.str) cir.sto conf)
+  :_  this(str str)
+  (send-chat-update [[%config cir.sto u.conf] str])
 ::
+::  +handle-rumor-circle-config-source: on source, subscribe and add to inbox
+::  on remove source, send delete and remove data from state
+::
+++  handle-rumor-circle-config-source
+  |=  sto=rumor-story:hall
+  ^-  (quip move _this)
+  ?>  ?=(%config -.sto)
+  ?>  ?=(%source -.dif.sto)
+  ?.  =(cir.sto [our.bol %inbox])
+    :: ignore when cir.sto is not inbox
+    [~ this]
+  =*  circ  cir.src.dif.sto
+  =/  wir  /circle/(scot %p hos.circ)/[nom.circ]/grams/0/config/group
+  ::  we've added a source to our inbox
+  ::
+  ?:  add.dif.sto
+    =.  str
+      %_  str
+        src.inbox  (~(put in src.inbox.str) src.dif.sto)
+        ::
+        configs
+          ?:  (~(has by configs.str) circ)
+            configs.str
+          (~(put by configs.str) circ ~)
+      ==
+    ::
+    =/  pax  /circle/[nom.circ]/grams/0/config/group
+    :_  this(str str)
+    :-  [ost.bol %peer wir [hos.circ %hall] pax]
+    (send-chat-update [[%inbox inbox.str] str])
+  ::
+  =.  src.inbox.str  (~(del in src.inbox.str) src.dif.sto)
+  ::  we've removed a source from our inbox
+  ::
+  =.  str
+    %=  str
+      inbox  inbox.str
+    ::
+      configs   (~(del by configs.str) circ)
+      messages  (~(del by messages.str) circ)
+      peers     (~(del by peers.str) circ)
+    ==
+  =/  fake=circle:hall
+    [our.bol (crip (weld (trip 'hall-internal-') (trip nom.circ)))]
+  ::
+  :_  this(str str)
+  ;:  weld
+    ^-  (list move)
+    ::  just forward the delete to our clients
+    ::
+    ?~  (~(get by configs.str) fake)
+      [ost.bol %pull wir [hos.circ %hall] ~]~
+    ::  if we get a delete from another ship, delete our fake circle copy
+    ::
+    :-  [ost.bol %pull wir [hos.circ %hall] ~]
+    [ost.bol %poke /f [our.bol %hall] [%hall-action [%delete nom.fake ~]]]~
```

Comment:
`:~` here also, like in the other comment.

Reply:
I don't want to change this. I prefer this style for this, given that the %delete action would be longer than 80 characters if it were indented further within a list.

Reply:
If I'm reading this correctly, can it be commuted to

```
	  ;:  weld
	    ::  forward the delete to our clients
	    ::
	    [ost.bol %pull wir [hos.circ %hall] ~]~
	    ::  if we got the delete from another ship, delete our fake circle copy
	    ::
	    ?~  (~(get by configs.str) fake)  ~
	    [ost.bol %poke /f [our.bol %hall] [%hall-action [%delete nom.fake ~]]]~
```